docs/folder-structure.md

```diff
@@ -0,0 +1,96 @@
+# F36 Folder Structure
+
+To make it easy to maintain our files and folders we ask every contributor to follow this folder structure.
+Every package has its own folder, except for components which should be kept in the folder `/components` (every component must have its own folder).
+
+## Table of contents
+
+- [Components](#components)
+  - [Packages with more than one component](#packages-with-more-than-one-component)
+  - [Packages with utility functions](#packages-with-utility-functions)
+  - [Why do we use MDX?](#why-do-we-use-mdx)
+  - [Why is every `.mdx` file named "README"](#why-is-every-mdx-file-named-readme)
+
+## Components
+
+The basic folder structure of a component looks like this:
+
+```
+/my-component
+  /src
+    index.ts
+    MyComponent.tsx
+    MyComponent.styles.ts
+  /stories
+    # stories for storybook
+  /test
+    # test files
+  README.mdx
+  package.json
+```
+
+### Packages with more than one component
+
+Sometimes the package will include a "base" component and some variants that are build from the base component (e.g. Button).
+In these cases, the folder structure of the package will look more like this:
+
+```
+/button
+  /src
+    /ButtonGroup
+      README.mdx
+      index.ts
+      ButtonGroup.tsx
+    /IconButton
+      README.mdx
+      index.ts
+      IconButton.tsx
+    /ToggleButton
+      README.mdx
+      index.ts
+      IconButton.tsx
+    index.ts
+    Button.tsx
+  /stories
+    # stories for storybook
+  /test
+    # test files
+  README.mdx
+  package.json
+```
+
+### Packages with utility functions
+
+Another possible case is that the package has a component and some utility functions. The folder structure will loke like this:
+
+```
+/datetime
+  /src
+    /RelativeDateTime
+      README.mdx
+      index.ts
+      RelativeDateTime.tsx
+    /utils
+      README.mdx
+      index.ts
+      formatDateTimeUtils.ts
+    index.ts
+    DateTime.tsx
+  /stories
+    # stories for storybook
+  /test
+    # test files
+  README.mdx
+  package.json
+```
+
+### Why do we use MDX?
+
+We use `.mdx` file format instead of just `.md` because these files are also used
+by our website to generate its pages.
+You can read more about the benefits of mdx [here](https://mdxjs.com/).
+
+### Why is every `.mdx` file named "README"?
+
+The markdown files must be named "README" because we want them to show up in GitHub
+for the people that navigate through the folders of the repo looking for documentation.
```

Comment:
hmm, now when I think about it, when I will look for the README for the button, I would need to go through the all REAME's that was found and check the file source to open the correct one. but I guess that's a common practice to call it README.mdx instead of Button.mdx, right?

Reply:
They need to be called "README" otherwise you won't see anything when you go here in Github
https://github.com/contentful/forma-36/tree/forma-v4/packages/components/button

Reply:
And, look at this:

<img width="1256" alt="Screenshot 2021-11-02 at 14 10 23" src="https://user-images.githubusercontent.com/6597467/139853009-77360201-6c21-4a81-a3cd-790ded55d5aa.png">